lib/src/segy.c

```diff
@@ -942,13 +942,20 @@ int segy_flush( segy_datasource* ds ) {
     return SEGY_OK;
 }
 
-static int segy_seek( segy_datasource* ds,
-                      int trace,
-                      long trace0,
-                      int trace_bsize ) {
-
-    trace_bsize += SEGY_TRACE_HEADER_SIZE;
-    long long pos = (long long)trace0 + (trace * (long long)trace_bsize);
+static int seek_traceheader_offset(
+    segy_datasource* ds,
+    int trace,
+    int traceheader,
+    long long offset,
+    long trace0,
+    int trace_bsize
+) {
+    long long trace_size = trace_bsize +
+                           SEGY_TRACE_HEADER_SIZE * ds->metadata.traceheader_count;
+    long long pos = trace0 +
+                    trace * trace_size +
+                    traceheader * SEGY_TRACE_HEADER_SIZE +
+                    offset;
```

Comment:
Do we need to check if `traceheader` is smaller than the number of trace headers? I could see that we start seeking inside the trace samples without noticing. 

Example:
The file has 3 trace headers per trace (standard trace header + 2 extended trace headers) and 1000 samples. Calling with `seek_trace_header = 3` would seek within for the header within the samples of the current trace.

1. Could this ever happen?
2. Do we safeguard against this or is this considered a user error.